### DISCUSSION TITLE
What's the story behind fpv_video from vc/src/hello_pi/hello_video/video.c?

### DISCUSSION THREAD

**Discussion by lida2003** [svpcom/wfb-ng#294] [Questions]
Well, I'm trying to understand the code. And what I have found is:

vc/src/hello_pi/hello_video/video.c is quite similar to fpv_video:

- three tunnels
> set_tunnel(tunnel, video_decode, 131, video_scheduler, 10);
> set_tunnel(tunnel+1, video_scheduler, 11, video_render, 90);
> set_tunnel(tunnel+2, clock, 80, video_scheduler, 12);
- data for video.c is file v.s. data for fpv_video is pipe

BUT I really can't understand what these numbers(131/10/11/90/80/12 for?

Any clue or links is well appreciated @svpcom 

[fpv_video_diff.zip](https://github.com/svpcom/wfb-ng/files/11016433/fpv_video_diff.zip) is the difference between `vc/src/hello_pi/hello_video/video.c` and fpv_video

**Comment by svpcom** [comment]
All these constants already were there (but on different lines). You can read OpenMAX standard for reference. It has structure similar to gstreamer. So most of the changes in fpv_video.c is refactoring of example from Broadcom.
